docs/core/testing/unit-testing-mstest-writing-tests.md

```diff
@@ -0,0 +1,396 @@
+---
+title: MSTest writing tests
+description: Learn about how to write tests using MSTest.
+author: Evangelink
+ms.author: amauryleve
+ms.date: 07/18/2024
+---
+
+# Writing tests with MSTest
+
+In this article, you will learn about the APIs and conventions used by MSTest to help you write and shape your tests.
+
+## Attributes
+
+MSTest uses custom attributes to identify and customize tests.
+
+To help provide a clearer overview of the testing framework, this section organizes the members of the <xref:Microsoft.VisualStudio.TestTools.UnitTesting> namespace into groups of related functionality.
+
+> [!NOTE]
+> Attribute elements, whose names end with "Attribute", can be used either with or without "Attribute" on the end and for parameterless constructors with or without parenthesis. For example, the following code examples function identically:
+>
+> `[TestClass()]`
+>
+> `[TestClassAttribute()]`
+>
+> `[TestClass]`
+>
+> `[TestClassAttribute]`
+
+### Attributes used to identify test classes and methods
+
+Every test class must have the `TestClass` attribute, and every test method must have the `TestMethod` attribute. For more information, see [Anatomy of a unit test](https://learn.microsoft.com/previous-versions/ms182517(v=vs.110)).
+
+#### TestClassAttribute
+
+The [TestClass](<xref:Microsoft.VisualStudio.TestTools.UnitTesting.TestClassAttribute>) attribute marks a class that contains tests and, optionally, initialize or cleanup methods.
+
+This attribute can be extended to update or extend the behavior.
+
+Example:
+
+```csharp
+[TestClass]
+public class MyTestClass
+{    
+}
+```
+
+#### TestMethodAttribute
+
+The [TestMethod](<xref:Microsoft.VisualStudio.TestTools.UnitTesting.TestMethodAttribute>) attribute is used inside a `TestClass` to define the actual test method to run.
+
+The method should be an instance method defined as `public void` or `public Task` (optionally `async`) and be parameterless.
```

Comment:
```suggestion
The method should be an instance method defined as `public void`, `public Task`, `public ValueTask`. It can optionally specify `async`.

The method should have zero parameters, unless it is used with `[DataRow]`, `[DynamicData]` or similar attribute that provides test case data to the test method.
```

Reply:
A true we added support for `ValueTask`, we should add the version fro which it's supported.

Reply:
```suggestion
The method should be an instance `public` method defined as `void`, `Task`, or `ValueTask` (starting with MSTest v3.3). It can optionally be `async` but should not be `async void`.

The method should have zero parameters, unless it is used with `[DataRow]`, `[DynamicData]` or similar attribute that provides test case data to the test method.
```